### PR TITLE
Fix float to bool cast

### PR DESCRIPTION
**Pull request by josel-amd**

### CODE REVIEW

**Review by mgehre-amd — CHANGES_REQUESTED** [review]
TOSA.cast seems to support the intended behavior for bool

File: test/mlir/conversion/onnx_to_tosa/Math/Elementwise.mlir [hunk 1]
```diff
@@ -26,6 +26,19 @@ func.func @test_cast_i8_i1(%arg0: tensor<1x21x1x1xi8>) -> tensor<1x21x1x1xi1> {
...
+
+// -----
+
+func.func @test_cast_f32_i1(%arg0: tensor<13x21x1xf32>) -> tensor<13x21x1xi1> {
+  %0 = "onnx.Cast"(%arg0) {to = i1} : (tensor<13x21x1xf32>) -> tensor<13x21x1xi1>
+  "func.return"(%0) : (tensor<13x21x1xi1>) -> ()
+
+   // CHECK-LABEL: func @test_cast_f32_i1
+   // CHECK-SAME:   (%[[VAL_0:.*]]: tensor<13x21x1xf32>) -> tensor<13x21x1xi1> {
+   // CHECK:        %[[VAL_1:.*]] = tosa.cast %arg0 : (tensor<13x21x1xf32>) -> tensor<13x21x1xi1>
```
Comment (TinaAMD) [inline test/mlir/conversion/onnx_to_tosa/Math/Elementwise.mlir:38]:
```suggestion
   // CHECK:        %[[VAL_1:.*]] = tosa.cast %[[VAL_0]] : (tensor<13x21x1xf32>) -> tensor<13x21x1xi1>
```

**Review by mgehre-amd — APPROVED** [review]
Thanks!

**Comment by mgehre-amd** [comment]
We should remember to merge this also to the release branch

**Comment by josel-amd** [comment]
> We should remember to merge this also to the release branch

Better to do it now before I forget. On it

**Review by TinaAMD — APPROVED** [review]
LGTM

**Pull request merged**